src/snapred/backend/data/LocalDataService.py

```diff
@@ -481,28 +482,37 @@ def writeNormalizationIndexEntry(self, entry: IndexEntry):
 
     def createNormalizationIndexEntry(self, request: CreateIndexEntryRequest) -> IndexEntry:
         indexer = self.normalizationIndexer(request.runNumber, request.useLiteMode)
-        return indexer.createIndexEntry(**request.model_dump())
+        entryParams = request.model_dump()
+        entryParams["version"] = entryParams.get("version", VersionState.NEXT)
+        if entryParams["version"] is None:
+            entryParams["version"] = VersionState.NEXT
+        return indexer.createIndexEntry(**entryParams)
 
     def createNormalizationRecord(self, request: CreateNormalizationRecordRequest) -> NormalizationRecord:
         indexer = self.normalizationIndexer(request.runNumber, request.useLiteMode)
         return indexer.createRecord(**request.model_dump())
 
     def normalizationExists(self, runId: str, useLiteMode: bool) -> bool:
-        version = self.normalizationIndexer(runId, useLiteMode).currentVersion()
+        version = self.normalizationIndexer(runId, useLiteMode).getLatestApplicableVersion(runId)
         return version is not None
 
     @validate_call
-    def readNormalizationRecord(self, runId: str, useLiteMode: bool, version: Optional[int] = None):
+    def readNormalizationRecord(self, runId: str, useLiteMode: bool, version: Optional[Version] = VersionState.LATEST):
         """
         Will return a normalization record for the given version.
         If no version given, will choose the latest applicable version from the index.
         """
         indexer = self.normalizationIndexer(runId, useLiteMode)
-        if version is None:
+        if version is VersionState.LATEST:
```

Comment:
In one part you use `VersionState.LATEST` to reference the current version, but here it is referencing the latest applicable version.  Maybe we need another value in `VersionState`?

Reply:
The conversion of latest to current was a mistake.  We will actually be trying to remove these intermediates entirely in the future